src/components/molecules/SpaceEditForm/SpaceEditForm.tsx

```diff
@@ -257,22 +255,22 @@ export const SpaceEditForm: React.FC<SpaceEditFormProps> = ({
 
   const { ownedVenues } = useOwnedVenues({});
 
-  const backButtonOptionList =
-    ownedVenues.reduce(
-      (
-        obj: Record<string, WithId<AnyVenue>> | null,
-        venue: WithId<AnyVenue>
-      ) => {
-        if (roomVenue?.worldId !== venue.worldId || venue.id === roomVenueId) {
-          return obj;
-        }
-
-        return { [venue.id]: venue, ...obj };
-      },
-      {}
-    ) ?? ALWAYS_EMPTY_OBJECT;
+  const backButtonOptionList = useMemo(
+    () =>
+      Object.fromEntries(
+        ownedVenues
+          .filter(({ id, worldId }) =>
+            roomVenue?.worldId !== worldId || id === roomVenueId ? false : true
```

Comment:
```suggestion
           ! ( roomVenue?.worldId !== worldId || id === roomVenueId )
```

or applying De Morgan's laws

```suggestion
             roomVenue?.worldId === worldId && id !== roomVenueId
```

Reply:
done 97a5880